### ISSUE TITLE
Unexpected behaviour when setting proxy

### ISSUE THREAD

**Issue by yarafan** [lostisland/faraday#786]
## Basic Info

* Faraday Version: 0.13.1
* Ruby Version: 2.3.1

## Issue description
Faraday ignores proxy that was set during initialize if no url was provided

## Steps to reproduce
```
irb(main):001:0> Faraday.new('http://www.example.com', :proxy => 'http://localhost:9999').get
Faraday::ConnectionFailed: Failed to open TCP connection to localhost:9999 (Connection refused - connect(2) for "localhost" port 9999)
        from (irb):1
```
works as expected but 
```
irb(main):002:0> Faraday.new(:proxy => 'http://localhost:9999').get("http://www.example.com")                                                                                                                              
=> #<Faraday::Response:0x00560d40ed9b28 @on_complete_callbacks=[], @env=#<Faraday::Env @method=:get @body="<html>\r\n<head><title>302 Found</title></head>\r\n<body bgcolor=\"white\">\r\n<center><h1>302 Found</h1></center>\r\n<hr><center>nginx/1.13.4</center>\r\n</body>\r\n</html>\r\n" @url=#<URI::HTTP http://test.com> @request=#<Faraday::RequestOptions (empty)> @request_headers={"User-Agent"=>"Faraday v0.13.1"} @ssl=#<Faraday::SSLOptions (empty)> @response=#<Faraday::Response:0x00560d40ed9b28 ...> @response_headers={"server"=>"nginx/1.13.4", "date"=>"Wed, 04 Apr 2018 12:41:03 GMT", "content-type"=>"text/html", "content-length"=>"161", "connection"=>"close", "location"=>"https://www.test.com/", "x-dis-request-id"=>"85d1d218a743924f541c943482176bd6"} @status=302 @reason_phrase="Moved Temporarily">>
```
ignores proxy settings.

Or am i doing something wrong?

**Comment by iMacTia** [comment]
Hi @yarafan, I think this was addressed in [v0.14.0](https://github.com/lostisland/faraday/releases/tag/v0.14.0) (see the first 2 PRs).
Could you please try to update Faraday to that version and see if the issue gets resolved?

**Comment by yarafan** [comment]
thanks for the help, it works as expected now
